plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergSplitSource.java

```diff
@@ -59,6 +61,9 @@ public IcebergSplitSource(CloseableIterable<CombinedScanTask> combinedScanIterab
         Iterator<FileScanTask> iterator = limit(fileScanIterator, maxSize);
         while (iterator.hasNext()) {
             FileScanTask task = iterator.next();
+            if (!task.deletes().isEmpty()) {
+                throw new TrinoException(NOT_SUPPORTED, "Iceberg tables with delete files are not supported");
```

Comment:
missed this earlier, a SchemaTableName would be good to add here in the error message.